lib/tracing-limit/benches/limit.rs

```diff
@@ -10,7 +10,8 @@ use std::{
     sync::{Mutex, MutexGuard},
 };
 use tracing::{field, span, Event, Id, Metadata};
-use tracing_limit::LimitSubscriber;
+use tracing_limit::Limit;
+use tracing_subscriber::layer::SubscriberExt;
 
 fn criterion_benchmark(c: &mut Criterion) {
     c.bench_function("basline_record", |b| {
```

Comment:
You could use a [`criterion::ParameterizedBenchmark`](https://bheisler.github.io/criterion.rs/book/user_guide/comparing_functions.html) to compare performance with and without rate-limiting across various numbers of events. That way, Criterion would output graph that compares rate-limiting and not rate-limiting side by side.